runtime/java/src/common/Node.java

```diff
@@ -121,6 +137,28 @@ public DecoratedNode decorate() {
 	 */
 	public abstract String getName();
 
+	public Object duplicate(Object redex, Object rule){

```

Comment:
What is the purpose of these methods?

Reply:
These are the noop implementations for origin tracking helper methods. These could probably be changed to panic since they should never be called since now the trackedness information is precise after moving it into `nonterminalType`. The semantics/use of (the specializations of) these helpers is discussed in https://github.com/melt-umn/melt-website/blob/feature/origintracking/silver/concepts/origins/index.md .

Reply:
So these are always overridden in the nonterminal/production subclasses?  Could these just be made abstract?

Reply:
Moved them to `TrackedNode` and made abstract.